pkg/storage/etcd/etcd_helper.go

```diff
@@ -72,18 +72,33 @@ type EtcdConfig struct {
 	CertFile   string
 	CAFile     string
 	Quorum     bool
+	Username   string
+	Password   string
 }
 
-func (c *EtcdConfig) newEtcdClient() (etcd.Client, error) {
+func (c *EtcdConfig) newClientConfig() (*etcd.Config, error) {
```

Comment:
IMHO, no need to return *etcd.Config here, just return etcd.Config, thus you have no need to construct etcd.Client like `etcd.New(*clientConfig)`